### ISSUE TITLE
[Accessibility]: Compare page: Focus lost when selecting a revision

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35532]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34426

### Problem description

Using a keyboard, select a revision like this:

<img width="804" alt="image" src="https://user-images.githubusercontent.com/9516420/168784823-0544a352-150b-4158-9647-a89240033f81.png">

Notice that focus is lost

### Expected behavior

Focus should be returned to the trigger that opened the selector (the button)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by felixfbecker** [comment]
Related: https://github.com/sourcegraph/sourcegraph/issues/35476

**Comment by lrhacker** [comment]
This appears to already be fixed/working as expected in the latest version of the application (recorded on S2).

https://user-images.githubusercontent.com/10523985/200889192-9257f0bc-d1cd-4122-bd02-d7a02498b3f5.mov
